toolchain/check/handle_impl.cpp

```diff
@@ -453,10 +453,7 @@ auto HandleParseNode(Context& context, Parse::ImplDefinitionStartId node_id)
       impl_decl_id, impl_info.scope_id,
       context.generics().GetSelfSpecific(impl_info.generic_id));
   StartGenericDefinition(context);
-
-  if (!impl_info.is_defined()) {
```

Comment:
Do we have tests with two `impl`, class, function, interface definitions with the same signature?

Reply:
See tests which mention "error: redefinition of"

But also, see the CHECK on line 446 here